ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -1012,6 +1012,48 @@ let on_notification server (notification : Client_notification.t) :
     in
     state
 
+module Log_config = struct
+  let parse_log_env_var var_name =
+    match String.split ~on:',' (Sys.getenv var_name) with
+    | value :: sections -> (Some value, sections)
+    | [] -> (None, [])
+    | exception Not_found -> (None, [])
+
+  (* note: merlin logs are buffered, so one has to wait until they see the
+     output; we could flush the buffer after each merlin command dispatch? *)
+  let with_merlin_log_file f =
+    let log_file, sections = parse_log_env_var "MERLIN_LOG" in
```

Comment:
Is there no way to have merlin write the ocamllsp log? It seems annoying to have to check two different files for this.

Reply:
> Is there no way to have merlin write the ocamllsp log?

Not without making changes to merlin, but, currently, we can pass `-` to both LOG variables to redirect logs to stderr; then we can have them in one place. 

From my experience, I found having separate log files quite convenient because of merlin log buffering and how much logs it can produce. We can evolve our logging as the need rises.

--

Also, merlin's buffering is a bit of annoyance for debugging; it would be great to possible flush the buffer after each `Document.dispatch` call. Wdyt?

Reply:
If you don't want merlin's log, can't you just disable it? Or add a marker that distinguishes the two types of logs and then filter in your log viewer? One of the most useful things about logs is the linearized output. Seeing it in a different file is going to reduce the ability to debug anything.

> Also, merlin's buffering is a bit of annoyance for debugging; it would be great to possible flush the buffer after each Document.dispatch call. Wdyt?

Sure, that sounds reasonable.

Reply:
But the issue with merlin not exposing its logging internals remains, so we can’t have merlin and olsp use the same file for logging except for stderr, which allows to have a single log. I can send a patch to merlin to fix this, but doesn’t have to stop this PR; shall a user like merlin and olsp to use the same file, they can pass the same file path as both env vars. Does this make sense?